proposal.rst

```diff
@@ -239,7 +337,99 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+**Bonding stage: Prior–May 28**
+
+* Get to know the community and interact with mentors.
+* Keep diving into Xapian and Omega's code base.
+* Fix some issues to go through the code review process.
+* Read more papers and have a clearer understanding of different click models.
+* Have clear blueprint of the project and think about the rough implementation of click model abstract class.
+
+**Coding Week 1: May 28–June 3**
+
+* Discuss the details of abstract class of click model with mentors.
+* Start implementing the abstract class.
+
+**Coding Week 2: June 4–June 10**
+
+* Final Week in unversity.
+* Finish implementing the abstract class.
+* Adapt the exsiting SimplifiedDBN class to the abstract class.
+* Start working on the DBN class with EM algorithm.
+
+**Coding Week 3: June 11–June 17**
+
+* Final Week in university.
+* Finish the training method for the DBN class.
+* Discuss the DCM model template with mentors.
+* Start writing automated test for the click models.
+* **Deliverable:** DBN model with EM alogorithm
+
+**Coding Week 4: June 18–June 24**
+
+* Obtain the Obtain the relevance judgements dataset for training the models.
+* Start testing the DBN models with click logs. 
+* Implementing the DCM model.
+
+**Coding Week 5: June 25–July 1**
+
+* GSoC Evaluation: June 25 - 29
+* Finish testing the DBN model.
+* Finish implementing the DCM model.
+
+* **Deliverable:** DCM model.
+
+**Coding Week 6: July 2–July 8**
+
+* Start discussing the UBM model with mentors.
+* Start implementing the UBM model.
+* Generate more click logs for testing.
+* Testing the DCM model.
+
+**Coding Week 7: July 9–July 15**
+
+* Finish implementing the UBM model.
+* Test the UBM model.
+* Obtain the relevance judgements dataset for training the models.
+* Obtain the predicted relevance judgments on the Omega click data.
+* **Deliverable:** UBM model.
+
+**Coding Week 8: July 16–July 22**
+
+* Implement the mechanism to automatically generate Qrel file.
+* Train the letor module on training file obtained from Omega click data.
+* Use the letor module for displaying relevant search results on top of SERP.
+
+**Coding Week 9: July 23–July 29**
+
+* GSoC Evaluation: July 23 - 27
+* Document various models and how to use the model.
+* Discuss the CTR evaluation with mentors.
+* **Deliverable:** a detailed documentation.
+
+**Coding Week 10: July 30–August 5**
+
+* Implementing the CTR evaluation method for click models.
+* Discuss the frequency of evaluation and the threshold of accuray to change the model.
+
+**Coding Week 11: August 5–August 11**
+
+* Documenting how to use the evaluation method for click models.
+* Decide the default click model and implement the machanism of changing the click models.
```

Comment:
I'd recommend implementing a way of selecting between click models when you implement the second click model (otherwise you end up with a bunch of code that cannot be used easily).

Reply:
That's true, I'll consider about the details of the mechanism.